pkg/wallet/wallet.go

```diff
@@ -439,7 +444,24 @@ func createAccountFromKeys(nickname string, privateKey []byte, publicKey []byte,
 	return &wallet, nil
 }
 
-func nicknameIsValid(nickname string) bool {
+func NicknameIsUnique(nickname string) (bool, error) {
+	// Load all accounts
+	wallets, err := LoadAll()
+	if err != nil {
+		return false, fmt.Errorf("loading wallets: %w", err)
+	}
+
+	// Check if nickname is unique
+	for _, wallet := range wallets {
+		if wallet.Nickname == nickname {
+			return false, nil
```

Comment:
maybe you should just return the error, and nil when the nickname is unique.
I would simplify your code because thats makes only 1 check to do then

Reply:
I pushed a commit to do this.